### PR TITLE
Fix value being decoded twice with `arrayFormat` option set to `separator`

### PR DESCRIPTION
**Pull request by cheqianxiao**
fix part of  #242  ( caused by #236

### CODE REVIEW

**Comment by sindresorhus** [comment]
> part of 242

What part does this not fix?

**Comment by sindresorhus** [comment]
You can use Array#includes instead of indexOf

**Comment by cheqianxiao** [comment]
> > part of 242
> 
> What part does this not fix?

it not fix : 
```js
parse(queryString, { arrayFormat: "comma", parseBooleans: true, parseNumbers: true });
// Request: id%28in%29=992232%2C992237%2C992240
// Parsed: 
// {
//    id(in): '992232,992237,992240'
// }
```

**Comment by sindresorhus** [comment]
Can you add a [failing test](https://github.com/avajs/ava/blob/master/docs/01-writing-tests.md#failing-tests) for that case that you're not fixing?

File: test/parse.js [hunk 1]
```diff
@@ -310,3 +310,14 @@ test('query strings having comma encoded and format option as `comma`', t => {
...
+
+test('value shoud not be decoded twice with arrayFormat set as `separator`', t => {
+	t.deepEqual(queryString.parse('foo=2020-01-01T00:00:00%2B03:00', {arrayFormat: 'separator'}), {
+		foo: '2020-01-01T00:00:00+03:00'
+	});
+});
+
+// See https://github.com/sindresorhus/query-string/pull/243#issuecomment-591179715
+test.failing('value separated by encoded comma will not be parsed as array with arrayFormat set as `comma`', t => {
+	t.fail();
```
Comment (sindresorhus) [inline test/parse.js:322]: This should actually be a valid test implementation, which would pass when the issue is fixed.

File: test/parse.js [hunk 2]
```diff
@@ -310,3 +310,14 @@ test('query strings having comma encoded and format option as `comma`', t => {
 		]
 	});
 });
+
+test('value shoud not be decoded twice with arrayFormat set as `separator`', t => {
```
Comment (sindresorhus) [inline test/parse.js:314]: Typo

**Pull request merged**